federation/pkg/dnsprovider/providers/google/clouddns/clouddns.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// clouddns is the implementation of pkg/dnsprovider interface for Google Cloud DNS
+package clouddns
+
+import (
+	"io"
+
+	"github.com/golang/glog"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/dns/v1"
+	"google.golang.org/cloud/compute/metadata"
+	"gopkg.in/gcfg.v1"
+
+	"k8s.io/kubernetes/federation/pkg/dnsprovider"
+	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns/internal"
+	"k8s.io/kubernetes/pkg/cloudprovider/providers/gce"
+)
+
+const (
+	ProviderName = "google-clouddns"
+	// operationPollInterval        = 3 * time.Second
+	// operationPollTimeoutDuration = 30 * time.Minute
+
+	// Each page can have 500 results, but we cap how many pages
+	// are iterated through to prevent infinite loops if the API
+	// were to continuously return a nextPageToken.
+	// maxPages = 25
```

Comment:
Are these commented out because these are not relevant to this package or because they are TBD? If former, it can just be removed. If latter, there should be a TODO and probably be removed as well.